### PR TITLE
Added operator+unsqueeze2 oneDNN fuse pass 

### PR DESCRIPTION
**Pull request by jakpiase**
### PR types
New features

### PR changes
OPs

### Describe
Added operator+unsqueeze2 oneDNN fuse pass. Relative performance gain on U2++ model should be around ~1%. Currently only transpose2 and elementwise_mul are supported in this pass. but the support may be extended in the future.

### CODE REVIEW

**Review by jczaja — COMMENTED** [review]
It seems there is no restriction on preceeding op, so why SetOutMemDescWithUnaqueeze2FuseSupport is only called from Transpose and Elementwise_xxx. How about FC+unsqueeze ?

**Review by Silv3S — DISMISSED** [review]
LGTM

File: paddle/fluid/framework/ir/mkldnn/operator_unsqueeze2_onednn_fuse_pass.cc [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+    bool has_axes_tensor =
+        std::find(names.begin(), names.end(), "AxesTensor") != names.end();
+    bool has_axes_tensor_list =
+        std::find(names.begin(), names.end(), "AxesTensorList") != names.end();
+
+    if (has_axes_tensor || has_axes_tensor_list) {
+      VLOG(3) << "Cannot fuse op with unsqueeze2, because its dims are "
+                 "specified either by AxesTensor or AxesTensorList and can be "
+                 "changed at runtime!";
```
Comment (Silv3S) [inline paddle/fluid/framework/ir/mkldnn/operator_unsqueeze2_onednn_fuse_pass.cc:78]: I think here should be `return` to cancel fuse under this condition. Currently we display warning and fuse anyway

Comment (jakpiase) [reply paddle/fluid/framework/ir/mkldnn/operator_unsqueeze2_onednn_fuse_pass.cc:78]: You're right, I have forgotten to add that

**Comment by jakpiase** [comment]
@jczaja as we have discussed offline, the restriction on preceding ops is listed in fuse pass file, so it will work properly

**Comment by jakpiase** [comment]
Restarting due to CheckPRTemplate not responding

**Review by zh794390558 — APPROVED** [review]
LGTM

**Pull request closed without merging**